src/components/Tutorial/__tests__/Tutorial.ts

```diff
@@ -0,0 +1,296 @@
+const lastThought = () => Array.from(document.querySelectorAll('[contenteditable="true"]')).at(-1)!
+
+/** Ensure we are at the given step of the tutorial. */
+const ensureStep = (step: number) => {
+  expect(store.getState().storageCache?.tutorialStep).toBe(step)
+}
+
+describe('Tutorial 1', async () => {
+  beforeEach(() => createTestApp({ tutorial: true }))
+  afterEach(cleanupTestEventHandlers)
+  afterAll(cleanupTestApp)
+  describe('step start', () => {
+    it('shows the welcome text', () => {
+      ensureStep(TUTORIAL_STEP_START)
+      expect(screen.getByText('Welcome to your personal thoughtspace.')).toBeInTheDocument()
+    })
+
+    it('shows the "Next" button, that leads to the first step', async () => {
+      await user.click(screen.getByText('Next'))
+      expect(screen.getByText(/First, let me show you how to create a new thought/)).toBeInTheDocument()
+    })
+  })
+
+  it('step first thought - how to create a thought ', async () => {
+    ensureStep(TUTORIAL_STEP_FIRSTTHOUGHT)
+    expect(screen.getByText('Hit the Enter key to create a new thought.')).toBeInTheDocument()
+    await user.keyboard('{Enter}')
+    expect(screen.getByText('You did it!')).toBeInTheDocument()
+
+    ensureStep(TUTORIAL_STEP_FIRSTTHOUGHT_ENTER)
+    expect(screen.getByText('Now type something. Anything will do.')).toBeInTheDocument()
+    await user.type(lastThought(), 'my first thought')
+    await user.keyboard('{Enter}')
+    await act(vi.runOnlyPendingTimersAsync)
+  })
+
+  it('step second thought - prompts to add another thought', async () => {
+    expect(screen.getByText('Well done!')).toBeInTheDocument()
+    expect(screen.getByText(/Try adding another thought/)).toBeInTheDocument()
+    ensureStep(TUTORIAL_STEP_SECONDTHOUGHT)
+
+    await user.type(lastThought(), 'my second thought')
+    await user.keyboard('{Enter}')
+    await act(vi.runOnlyPendingTimersAsync)
+
+    ensureStep(TUTORIAL_STEP_SECONDTHOUGHT_ENTER)
+
+    expect(screen.getByText('Good work!')).toBeInTheDocument()
+    expect(screen.getByText('Now type some text for the new thought.')).toBeInTheDocument()
+    await user.type(lastThought(), 'third thought')
+    await user.keyboard('{Enter}')
+    await act(vi.runOnlyPendingTimersAsync)
+  })
+
+  it('step subthought - how to create a thought inside thought', async () => {
+    ensureStep(TUTORIAL_STEP_SUBTHOUGHT)
+    expect(exportContext(store.getState(), [HOME_TOKEN], 'text/plain')).toBe(
+      `- __ROOT__
+  - my first thought
+  - my second thought
+  - third thought
+  - `,
+    )
+
+    // Now I am going to show you how to add a thought within another thought.
+    expect(screen.getByText(/Now I am going to show you how to add a thought/)).toBeInTheDocument()
+
+    // since we have cursor on empty thought
+    expect(screen.getByText(/Hit the Delete key to delete the current blank thought/)).toBeInTheDocument()
+    expect(screen.getByText(/Then hold the Ctrl key and hit the Enter key/)).toBeInTheDocument()
```

Comment:
Let's stick with a single `screen.getByText` for a given point in the tutorial. Testing the minimal unique text at the given step will give us effectively the same test coverage, but will be less maintenance if the tutorial text is changed in the future.

Likewise in similar occurrences below.

Reply:
@snqb Still needs attention